app/main/views/services.py

```diff
@@ -702,3 +709,106 @@ def set_remove_last_subsection_error_message():
                 service_id=service_id,
                 confirm_remove=None
                 ))
+
+
+@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/previous-services', methods=['GET'])
+@login_required
+def list_previous_services(framework_slug, lot_slug):
+    framework, lot = get_framework_and_lot_or_404(data_api_client, framework_slug, lot_slug)
+    if framework['status'] != 'open':
+        abort(404)
+
+    copy_all = request.args.get('copy_all', None)
```

Comment:
Have the query param part of this be hyphenated would be more consistent with the rest of the URL. Not a big deal if you want to leave this as is, but IMO it'd be nicer as `copy-all`. Your call.

Reply:
Weirdness about having a hyphen in the key for the parameter to `url_for()` (causes a Jinja2 syntax error) so gonna leave for now.